pkg/config/config.go

```diff
@@ -469,6 +469,11 @@ type ImageConfig struct {
 	// PauseCommand is the path of the binary we run in an infra
 	// container that's been instantiated using PauseImage.
 	PauseCommand string `toml:"pause_command"`
+	// PinnedImages is a list of container images that should be pinned
+	// and not subject to garbage collection by kubelet.
+	// Pinned images will remain in the container runtime's storage until
+	// they are manually removed. Default value: empty list (no images pinned)
+	PinnedImages []string `toml:"pinned_images"`
```

Comment:
We should mention in the docs that those are regular expressions and how it behaves. Having examples would be great as well.

Reply:
The names of the images provided under [pinned_image] are not exactly regular expressions. While users may enter patterns that resemble regular expressions, our code is designed to handle them in a suitable manner. This is mainly a backend logic and not a user-facing change.

Reply:
How about this one?
```
The `pinned_images` field allows specifying image names using either exact, glob, or keyword patterns. Exact matches must match the entire name, glob matches can have a wildcard * at the end, and keyword matches can have wildcards on both ends.
```

Reply:
Sounds good!